### PR TITLE
Improve Bluetooth and Account interactions

### PR DESCRIPTION
**Pull request by Supereg**
# Improve Bluetooth and Account interactions

## :recycle: Current situation & Problem
The latest SpeziBluetooth and SpeziDevices releases fixed several issues (https://github.com/StanfordSpezi/SpeziDevices/pull/9, https://github.com/StanfordSpezi/SpeziBluetooth/pull/46) including the duplicate delivery of measurements and minor pairing issues.
Additionally, this PR upgrades to the upcoming SpeziAccount release that provides several improvements when interacting with Account Services. We updated code to avoid directly interaction with FirebaseAuth.

## :gear: Release Notes 
* Fixed an issue where measurements were delivered twice or not at all.
* Fixed an issue where a Weight Scale was considered paired to early.
* Improved general structure of the application integrating latest SpeziAccount features.

## :books: Documentation
--

## :white_check_mark: Testing
--

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: ENGAGEHF.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1693,7 +1691,7 @@
 					"@executable_path/Frameworks",
 				);
 				MARKETING_VERSION = 1.0;
-				PRODUCT_BUNDLE_IDENTIFIER = edu.stanford.bdh.engagehf;
+				PRODUCT_BUNDLE_IDENTIFIER = edu.stanford.bdh.engagehf2;
```
Comment (PSchmiedmayer) [inline ENGAGEHF.xcodeproj/project.pbxproj:1694]: TODO to revert this before the merge.

File: ENGAGEHF.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1667,12 +1665,12 @@
...
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
-				CODE_SIGN_ENTITLEMENTS = "ENGAGEHF/Supporting Files/ENGAGEHF.entitlements";
+				CODE_SIGN_ENTITLEMENTS = "";
 				CODE_SIGN_IDENTITY = "Apple Development";
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_ASSET_PATHS = "";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 484YT3X9X7;
```
Comment (PSchmiedmayer) [inline ENGAGEHF.xcodeproj/project.pbxproj:1673]: TODO to revert this before the merge.

File: ENGAGEHF.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1911,8 +1909,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/StanfordSpezi/SpeziFirebase.git";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = 1.1.4;
+				branch = "feature/spezi-account-2.0";
```
Comment (PSchmiedmayer) [inline ENGAGEHF.xcodeproj/project.pbxproj:1912]: TODO to use the tagged release after the PR is merged.

File: ENGAGEHF/Account/InvitationCodeModule.swift [hunk 4]
```diff
@@ -7,34 +7,44 @@
...
+    func signOutAccount() async {
         do {
-            try Auth.auth().signOut()
+            try await _signOutAccount()
         } catch {
             logger.debug("Failed to sing out firebase account: \(error)")
         }
     }
 
+    private func _signOutAccount() async throws {
```
Comment (PSchmiedmayer) [inline ENGAGEHF/Account/InvitationCodeModule.swift:38]: Can we package this into the function above?

File: ENGAGEHF/Contacts/Contacts.swift [hunk 5]
```diff
@@ -6,6 +6,7 @@
 // SPDX-License-Identifier: MIT
 //
 
+import Contacts
```
Comment (PSchmiedmayer) [inline ENGAGEHF/Contacts/Contacts.swift:9]: Do we need that with the fix in SpeziContact?

File: ENGAGEHF/ENGAGEHFDelegate.swift [hunk 6]
```diff
@@ -24,25 +24,21 @@ import SwiftUI
...
-                if FeatureFlags.useFirebaseEmulator {
-                    FirebaseAccountConfiguration(
-                        authenticationMethods: [.emailAndPassword, .signInWithApple],
-                        emulatorSettings: (host: firestoreHost, port: 9099)
-                    )
-                } else {
-                    FirebaseAccountConfiguration(authenticationMethods: [.emailAndPassword, .signInWithApple])
-                }
-                FirestoreAccountStorage(storeIn: Firestore.userCollection)
+                AccountConfiguration(
```
Comment (PSchmiedmayer) [inline ENGAGEHF/ENGAGEHFDelegate.swift:29]: Nice to see this simplified in here!

File: ENGAGEHF/ENGAGEHFStandard.swift
Comment (PSchmiedmayer) [inline ENGAGEHF/ENGAGEHFStandard.swift:1]: Nice to see this simplified in here!

File: ENGAGEHF/Education/VideoManager.swift [hunk 7]
```diff
@@ -34,24 +35,27 @@ final class VideoManager: Module, EnvironmentAccessible, DefaultInitializable {
...
-        
-        authStateDidChangeListenerHandle = Auth.auth().addStateDidChangeListener { [weak self] _, user in
-            if let user {
-                Task {
-                    guard let collections = await self?.getVideoSections(user: user) else {
+
+        if let accountNotifications {
+            notificationTask = Task.detached { @MainActor [weak self] in
+                for await event in accountNotifications.events {
+                    guard let self else {
```
Comment (PSchmiedmayer) [inline ENGAGEHF/Education/VideoManager.swift:42]: Should we also check for Task cancellation in here? Applies to the other usages of this as well.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for pulling the SpeziAccount update into the app; the code changes look great!

File: ENGAGEHF/Education/VideoManager.swift (hunk 7 above)
Comment (Supereg) [reply ENGAGEHF/Education/VideoManager.swift:42]: I think the AsyncStream automatically checks for cancellation and just completes, if I’m not wrong. But I’ll re-review that 👍

File: ENGAGEHF.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1995,12 +1993,20 @@
...
+				minimumVersion = "2.0.0-beta.1";
+			};
+		};
 		A9623C022C17A3F500189BA1 /* XCRemoteSwiftPackageReference "SpeziDevices" */ = {
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/StanfordSpezi/SpeziDevices.git";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = 1.1.0;
+				branch = "fix/bump-xctest-extensions";
```
Comment (Supereg) [inline ENGAGEHF.xcodeproj/project.pbxproj:2008]: Waiting for bump

**Pull request merged**